### PR TITLE
1039 dapricot scale updated

### PR DESCRIPTION
**Pull request by yusong-tan**
Updated dapricot_patch.py, which needs to be moved to armory-private. Also updated the baseline model to accept different batch size inputs and added the RobustDPatch attack scenario config.

### CODE REVIEW

File: armory/art_experimental/attacks/dapricot_patch.py [hunk 1]
```diff
@@ -1,23 +1,673 @@
...
+            )
+
+        transformations = dict()
+        x_copy = x.copy()
+        patch_copy = patch.copy()
+        x_patch = x.copy()
+
+        if channels_first:
+            x_copy = np.transpose(x_copy, (0, 2, 3, 1))
+            x_patch = np.transpose(x_patch, (0, 2, 3, 1))
```
Comment (lcadalzo) [inline armory/art_experimental/attacks/dapricot_patch.py]: `x_patch` as defined by `x_patch = np.transpose(x_patch, (0, 2, 3, 1))` will not be used as this variable is then reset to an empty list. Should this line be removed?

Comment (yusong-tan) [reply armory/art_experimental/attacks/dapricot_patch.py]: Wouldn't this be used if the model is Pytorch?

File: armory/art_experimental/attacks/dapricot_patch.py [hunk 2]
```diff
@@ -1,23 +1,673 @@
...
+            raise ValueError(
+                "Number of images should be equal to the number of targets"
+            )
+
+        transformations = dict()
+        x_copy = x.copy()
+        patch_copy = patch.copy()
+        x_patch = x.copy()
+
+        if channels_first:
```
Comment (lcadalzo) [inline armory/art_experimental/attacks/dapricot_patch.py]: I think I may see an issue with this code if channels_first is True:

The value that gets passed for the `channels_first` arg is `self.estimator.channels_first`. Regardless of whether this is True, the `x` will always be channels_last when using Armory for dataset loading. So if `self.estimator.channels_first` is True, then we'll be calling `np.transpose(x_copy, (0, 2, 3, 1))` when `x_copy` is of shape `(3, 1008, 756, 3)` which will lead to an array of shape `(3, 756, 3, 1008)` which would be a problem

Comment (yusong-tan) [reply armory/art_experimental/attacks/dapricot_patch.py]: I see. `self.estimator.channels_first == True` implies the model assumes the data has channel first, which isn't the case for the DAPRICOT dataset, or any dataset from Armory for that matter. The burden is on the model to transpose the data correctly and not on ART.

File: armory/art_experimental/attacks/dapricot_patch.py (hunk 1 above)
Comment (yusong-tan) [reply armory/art_experimental/attacks/dapricot_patch.py]: We should then remove any code under `if channels_first:`

File: armory/art_experimental/attacks/dapricot_patch.py (hunk 2 above)
Comment (lcadalzo) [reply armory/art_experimental/attacks/dapricot_patch.py]: correct

File: armory/art_experimental/attacks/dapricot_patch.py (hunk 1 above)
Comment (lcadalzo) [reply armory/art_experimental/attacks/dapricot_patch.py]: My point here was just that `x_patch` is then, regardless, immediately set to an empty list in [this](https://github.com/yusong-tan/armory/blob/1039-dapricot-scale_ytan/armory/art_experimental/attacks/dapricot_patch.py#L518) line, making [this](https://github.com/yusong-tan/armory/blob/1039-dapricot-scale_ytan/armory/art_experimental/attacks/dapricot_patch.py#L514) line irrelevant. 

But I think you're right in that we should remove code under `if channels_first`, considering that Armory always loads data in channels_last format.

Comment (yusong-tan) [reply armory/art_experimental/attacks/dapricot_patch.py]: I agree. I did not see which line you were referring to when you said "then reset to an empty list," but now I know.

**Pull request merged**